.github/workflows/testAndPublish.yml

```diff
@@ -0,0 +1,243 @@
+name: Run tests and publish
+
+on:
+  push:
+    branches:
+    - master
+    - beta
+    - rc
+    - 'try-**'
+    tags:
+    - 'release-**'
+
+  pull_request:
+    branches:
+    - master
+    - beta
+    - rc
+
+  workflow_dispatch:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+env:
+  START_BUILD_NUMBER: 100000
+  pullRequestNumber: ${{ github.event_name == 'pull_request' && github.event.number }}
+  scons_publisher: NV Access
+  # Comment out any of the feature_* variables to disable the respective build feature.
+  # They are checked for existence of content, not specific value.
+  # feature_buildSymbols: configured
+  # feature_uploadSymbolsToMozilla: configured
+  # feature_buildAppx: configured
+  # feature_crowdinSync: configured
+  # feature_signing: configured
+
+defaults:
+  run:
+    shell: cmd
+
+jobs:
+  buildNVDA:
+    name: Build NVDA
+    runs-on: windows-latest
+    steps:
+    - name: Checkout NVDA
+      uses: actions/checkout@v4
+      with:
+        submodules: true
+    - name: Install dependencies
+      uses: actions/setup-python@v5
+      with:
+        python-version: '3.11'
+        architecture: x86
+    - name: Cache dependencies
+      uses: actions/cache@v4
+      with:
+        path: .venv
+        key: ${{ runner.os }}-${{ hashFiles('requirements.txt') }}
+    - name: Set version variables
+      run: powershell.exe ci/scripts/setBuildVersionVars.ps1
+    - name: Set scons args
+      run: powershell.exe ci/scripts/setSconsArgs.ps1
+    - name: Prepare source code
+      run: scons source %sconsArgs% --all-cores
```

Comment:
`--all-cores` will mangle the output. Didn't we consider this problematic before?

Reply:
The main reason we avoided it is appveyor runners didn't have multiple real cores, only virtual ones.
We should probably consider making them linear for verbose debug builds, which you can trigger by rerunning a failing build

Reply:
I've made use of `runner.debug` for this